src/_P021_Level.ino

```diff
@@ -244,19 +244,20 @@ boolean Plugin_021(uint8_t function, struct EventStruct *event, String& string)
 
     case PLUGIN_WEBFORM_LOAD:
     {
-      const bool extFunc = bitRead(P021_FLAGS, P021_EXT_FUNCT); // Extended functionality selected
+      const uint16_t flags   = P021_FLAGS;                     // shortcut to read existing configuration flags
+      const bool     extFunc = bitRead(flags, P021_EXT_FUNCT); // Extended functionality selected
 
        # ifdef PLUGIN_021_DEBUG
       {
         const P021_control_state control_state = (P021_control_state)UserVar.getFloat(event->TaskIndex, P021_VALUE_STATE);
 
         // Add some debug information
         const String outpstring = (int)UserVar.getFloat(event->TaskIndex, P021_VALUE_OUTPUT) == 1 ? F("on") : F("off");
-        String msg        = strformat(F("State= %s, Output= %s, Remote= %d, Timer= %d sec"),
-                                      P021_printControlState(control_state),
-                                      outpstring.c_str(),
-                                      P021_remote[event->TaskIndex],
-                                      millis2seconds(timePassedSince(P021_timestamp[event->TaskIndex])));
+        String msg              = strformat(F("State= %s, Output= %s, Remote= %d, Timer= %d sec"),
```

Comment:
Assigning the result of `strformat` to a non-const varialble will result in both more memory usage and larger build file compared to when you call it directly where you use it. (assuming you only use it once)

```c++
addFormNote(strformat(...));
```
This will not create a copy of the string on the stack, but directly feed it to the `addFormNote` function.
After the function returns the temporary created String object is destructed and thus not taking up any memory anymore.